apps/condo/domains/contact/hooks/useImporterFunctions.tsx

```diff
@@ -0,0 +1,135 @@
+
+const SPLIT_PATTERN = /[, ;.]+/
+
+const parsePhones = (phones: string) => {
+    const splitPhones = phones.split(SPLIT_PATTERN)
+    return splitPhones.map(phone => {
+        if (phone.startsWith('8')) {
+            phone = '+7' + phone.substring(1)
+        }
+        phone = phone.replace(/[^0-9+]/g, '')
+        return normalizePhone(phone)
+    })
+}
+
+export const useImporterFunctions = (): [Columns, RowNormalizer, RowValidator, ObjectCreator] => {
+    const userOrganization = useOrganization()
+    const client = useApolloClient()
+    const { addressApi } = useAddressApi()
+
+    const userOrganizationId = get(userOrganization, ['organization', 'id'])
+
+    // @ts-ignore
+    const contactCreateAction = Contact.useCreate({},
+        () => Promise.resolve())
+
+    const columns: Columns = [
+        { name: 'Address', type: 'string', required: true },
+        { name: 'Unit Name', type: 'string', required: true },
+        { name: 'Phones', type: 'string', required: true },
+        { name: 'Full Name', type: 'string', required: true },
+        { name: 'Email', type: 'string', required: false },
+    ]
+
+    const contactNormalizer: RowNormalizer = (row) => {
+        const addons = { address: null, property: null, phones: null, fullName: null, email: null }
+        if (row.length !== columns.length) return Promise.resolve({ row })
+        const [address, , phones, fullName, email] = row
+        email.value = email.value && String(email.value).trim().length ? String(email.value).trim() : undefined
+        return addressApi.getSuggestions(String(address.value)).then(result => {
+            const suggestion = get(result, ['suggestions', 0])
+            if (suggestion) {
+                addons.address = suggestion.value
+                const where = {
+                    address_contains_i: suggestion.value,
+                    organization: { id: userOrganizationId },
+                }
+                // TODO (savelevMatthew): better way to detect building
+                return searchProperty(client, where, undefined).then((res) => {
+                    addons.property = res.length > 0 ? res[0].value : null
+                    addons.phones = parsePhones(String(phones.value))
+                    addons.fullName = String(fullName.value).trim()
+                    addons.email = normalizeEmail(email.value)
+                    return { row, addons }
+                })
+            }
+            addons.phones = parsePhones(String(phones.value))
+            addons.fullName = String(fullName.value).trim()
+            addons.email = normalizeEmail(email.value)
+            return { row, addons }
+        })
+    }
+
+    const contactValidator: RowValidator = (row) => {
+        if (!row || !row.addons) return Promise.resolve(false)
+        if (!row.addons.property) return Promise.resolve(false)
+        if (!row.addons.fullName) return Promise.resolve(false)
+
+        const rowEmail = row.row[4].value
```

Comment:
Use get from lodash. It's possible to get exception in runtime if there will be no element in row[4]

Reply:
It's impossible. Since we require all rows in line 46. No 5th element on line 46 = no addons returning = resolving false in line 74

Reply:
This function is pure. I got no static typo or row length validation here in contactValidator. There is the case when some one of other developers can reuse your solution and work with this later, and then, if contactNormalizer will be change this function can throw an unhandled rejection)

Reply:
Ok, I'll fix it 👍